superagi/jobs/agent_executor.py

```diff
@@ -107,6 +94,30 @@
             session.close()
             engine.dispose()
 
+    def __execute_workflow_step(self, agent, agent_config, agent_execution_id, agent_workflow_step, memory,
+                                model_api_key, organisation, session):
+        logger.info("Executing Workflow step : ",agent_workflow_step.action_type)
+        if agent_workflow_step.action_type == "TOOL":
+            tool_step_handler = AgentToolStepHandler(session,
+                                                     llm=get_model(model=agent_config["model"], api_key=model_api_key,
+                                                                   organisation_id=organisation.id)
+                                                     , agent_id=agent.id, agent_execution_id=agent_execution_id,
+                                                     memory=memory)
+            tool_step_handler.execute_step()
+        elif agent_workflow_step.action_type == "ITERATION_WORKFLOW":
+            iteration_step_handler = AgentIterationStepHandler(session,
+                                                               llm=get_model(model=agent_config["model"],
+                                                                             api_key=model_api_key,
+                                                                             organisation_id=organisation.id)
+                                                               , agent_id=agent.id,
+                                                               agent_execution_id=agent_execution_id, memory=memory)
```

Comment:
The action type 'WAIT_STEP' is hardcoded. Consider using a constant or enum type to avoid hardcoding and improve maintainability.

Reply:
Yes using enum to handle agent_execution status